### PR TITLE
Core: Remove some logging constructors

### PR DESCRIPTION
**Pull request by nik9000**
Remove a few of the logger constructors that aren't widely used or
aren't used at all and deprecate a few more logger constructors in favor
of log4j2's `LogManager`.

### CODE REVIEW

File: qa/evil-tests/src/test/java/org/elasticsearch/common/logging/EvilLoggerTests.java [hunk 1]
```diff
@@ -298,8 +299,8 @@ public void testFindAppender() throws IOException, UserException {
     public void testPrefixLogger() throws IOException, IllegalAccessException, UserException {
         setupLogging("prefix");
 
-        final String prefix = randomBoolean() ? null : randomAlphaOfLength(16);
-        final Logger logger = Loggers.getLogger("prefix", prefix);
```
Comment (nik9000) [inline qa/evil-tests/src/test/java/org/elasticsearch/common/logging/EvilLoggerTests.java]: We plan to drop support for `null` here altogether.

File: server/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java [hunk 2]
```diff
@@ -345,10 +345,7 @@ static void init(
             if (foreground && maybeConsoleAppender != null) {
                 Loggers.removeAppender(rootLogger, maybeConsoleAppender);
             }
-            Logger logger = Loggers.getLogger(Bootstrap.class);
-            if (INSTANCE.node != null) {
-                logger = Loggers.getLogger(Bootstrap.class, Node.NODE_NAME_SETTING.get(INSTANCE.node.settings()));
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java]: We get the node name by default here.

**Review by rjernst — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/index/analysis/ESWordnetSynonymParser.java [hunk 3]
```diff
@@ -20,19 +20,18 @@
...
 import org.apache.lucene.analysis.Analyzer;
 import org.apache.lucene.analysis.synonym.WordnetSynonymParser;
 import org.apache.lucene.util.CharsRef;
 import org.apache.lucene.util.CharsRefBuilder;
-import org.elasticsearch.common.logging.Loggers;
 
 import java.io.IOException;
 
 public class ESWordnetSynonymParser extends WordnetSynonymParser {
+    private static final Logger logger = LogManager.getLogger(ESSolrSynonymParser.class);
```
[the remark this answers is not in the document]
Comment (nik9000) [reply server/src/main/java/org/elasticsearch/index/analysis/ESWordnetSynonymParser.java]: I made a mistake.

**Pull request merged**